### PR TITLE
Add publishing extensions tutorial. Fix #13 .

### LINKED ISSUE #13: Add "How to publish extensions" section to the extensions tutorial

### PR DESCRIPTION
**Pull request by oruebel**

### CODE REVIEW

**Comment by bendichter** [comment]
The info-graphic is cool. Do you think GitHub should be a part of it? Maybe it could replace git?

**Comment by oruebel** [comment]
> Do you think GitHub should be a part of it? Maybe it could replace git?

The main reason I used the git logo instead of the GitHub logo here is because (in contrast to the NDX-Catalog and PyPI) we don't really assume a particular hosting service for the Git repo of extensions. I.e., it doesn't really matter whether a developer uses GitHub, BitBucket, GitLab etc. to share their source code. However, this is a just a "info-graphic" so we should show what is most intuitive and ultimately (at least right now) GitHub is the main hosting service used to share extension repos. Long story short, I'm happy to change the Git logo to GitHub logo since that is probably more intuitive. I'll make the change and upload the new graphic in a bit.

File: docs/source/extensions_tutorial/7_publishing_extensions.rst [hunk 1]
```diff
@@ -1,4 +1,114 @@
...
 
-Neurodata extensions can be shared with the community using the `NDX Catalog <https://nwb-extensions.github.io/>`_
\ No newline at end of file
+.. image:: ../img/publishing_extensions.png
+   :width: 315px
+   :class: align-right
+
+Neurodata extensions can be shared with the community using the :ndx-catalog:`NDX Catalog <>`.
+As illustrated in the figure, the publication process is divided into three main steps:
+1) open release of the sources to the community using git, 2) open access of versioned
```
Comment (bendichter) [inline docs/source/extensions_tutorial/7_publishing_extensions.rst:10]:
```suggestion
1) open release of the sources to the community using GitHub, 2) open access of versioned
```

I would argue that git alone does not open the source code to the community

File: docs/source/extensions_tutorial/7_publishing_extensions.rst [hunk 2]
```diff
@@ -1,4 +1,114 @@
...
+
+1. Before publishing your licence online you should add a license file. Permissive licenses should be used if possible. A `BSD license <https://opensource.org/licenses/BSD-3-Clause>` is recommended.
+2. Modify ``README.md`` at the root directory of your extension repo to describe the extension for interested developers and users.
+3. The first step to publishing your extension then is to make your Git repository accessible online
+   via GitHub, or any other public Git hosting service of your choice. To publish your extension
+   on GitHub you will need a `GitHub account <https://github.com/>`_ and follow the following
+   `instructions to add an existing project to GitHub <https://help.github.com/en/articles/adding-an-existing-project-to-github-using-the-command-line>`
+4. Make a release for the extension on GitHub with the version number specified. e.g. if the version
+   is ``0.1.0``, then this page should exist: https://github.com/<my_username>/<my_extension>/releases/tag/0.1.0``.
+   See the `creating a release guide <https://help.github.com/en/github/administering-a-repository/creating-releases>` on GitHub for instructions on how to make a release on GitHub.
```
Comment (bendichter) [inline docs/source/extensions_tutorial/7_publishing_extensions.rst:25]: recommend semantic versioning

File: docs/source/extensions_tutorial/7_publishing_extensions.rst [hunk 3]
```diff
@@ -1,4 +1,114 @@
...
+6. Add and commit your changes to Git and push your changes to GitHub:
+
+.. code-block::
+
+    cd staged-extensions
+    git add <my_extension>
+    git commit -m "Add new catalog entry for <my_extension>"
+    git push
+
+7. Open a pull request. Building of your extension will be tested on Windows, Mac, and Linux.
```
Comment (bendichter) [inline docs/source/extensions_tutorial/7_publishing_extensions.rst:87]: This step might be confusing to users who haven't had experience with github. Is there a link we can provide?

File: docs/source/extensions_tutorial/7_publishing_extensions.rst [hunk 4]
```diff
@@ -1,4 +1,114 @@
...
+
+Releasing your extension on GitHub
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Before publishing your licence online you should add a license file. Permissive licenses should be used if possible. A `BSD license <https://opensource.org/licenses/BSD-3-Clause>` is recommended.
+2. Modify ``README.md`` at the root directory of your extension repo to describe the extension for interested developers and users.
+3. The first step to publishing your extension then is to make your Git repository accessible online
+   via GitHub, or any other public Git hosting service of your choice. To publish your extension
+   on GitHub you will need a `GitHub account <https://github.com/>`_ and follow the following
+   `instructions to add an existing project to GitHub <https://help.github.com/en/articles/adding-an-existing-project-to-github-using-the-command-line>`
```
[the remark this answers is not in the document]
Comment (oruebel) [reply docs/source/extensions_tutorial/7_publishing_extensions.rst:22]: The ndx-template creates a Git repository on your local computer, i.e,. while the template originates from a GitHub repo, the Git repo that it creates is not tied to GitHub. It is up to user to add the repo to GitHub (or any other hosting service they may choose)

File: docs/source/extensions_tutorial/7_publishing_extensions.rst (hunk 2 above)
Comment (oruebel) [reply docs/source/extensions_tutorial/7_publishing_extensions.rst]: I added a note to link to versioning guidelines

File: docs/source/extensions_tutorial/7_publishing_extensions.rst (hunk 3 above)
Comment (oruebel) [reply docs/source/extensions_tutorial/7_publishing_extensions.rst]: I added a corresponding link to GitHub docs. I also added a screenshot to show how to clone and fork repos.

**Comment by oruebel** [comment]
@bendichter thanks for the helpful comment. I believe I have addressed all your suggestion. Please take another look when you get a chance.

**Review by bendichter — APPROVED** [review]
looks great!

**Pull request merged**